lib/rules/require-numeric-literals.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Requires use of binary, hexadecimal, and octal literals instead of parseInt.
+ *
+ * Type: `Boolean`
+ *
+ * Value: `true`
+ *
+ * Version: `ES6`
+ *
+ * #### Example
+ *
+ * ```js
+ * "requireNumericLiterals": true
+ * ```
+ *
+ * ##### Valid
+ *
+ * ```js
+ * 0b111110111 === 503;
+ * 0o767 === 503;
+ * 0x1F7 === 503;
+ * ```
+ *
+ * ##### Invalid
+ *
+ * ```js
+ * parseInt("111110111", 2) === 503;
+ * parseInt("767", 8) === 503;
+ * parseInt("1F7", 16) === 255;
+ * ```
+ */
+
+var assert = require('assert');
+
+module.exports = function() {};
+
+module.exports.prototype = {
+    configure: function(options) {
+        assert(
+            options === true,
+            this.getOptionName() + ' option requires a true value or should be removed'
+        );
+    },
+
+    getOptionName: function() {
+        return 'requireNumericLiterals';
+    },
+
+    check: function(file, errors) {
+        var radixMap = {
```

Comment:
Unnecessary object creation per file checked. Move to this.radixMap in configure.